src/gitops/components/rollout/revisions/ReplicaSetInfo.ts

```diff
@@ -0,0 +1,215 @@
+import { PodKind } from 'src/components/topology/console/types';
+
+import {
+  K8sResourceCommon,
+  Operator,
+  OwnerReference,
+  Selector,
+} from '@openshift-console/dynamic-plugin-sdk';
+
+import { AnalysisRunKind } from '../model/AnalysisRunModel';
+import { RolloutKind } from '../model/RolloutModel';
+
+const annotationRevisionKey = 'rollout.argoproj.io/revision';
+const labelPodTemplateHashKey = 'rollouts-pod-template-hash';
+
+export enum ReplicaSetStatus {
+  Active,
+  Canary,
+  Preview,
+  Stable,
+}
+
+export type ImageInfo = {
+  name: string;
+  image: string;
+};
+
+export type AnalysisRunInfo = {
+  name: string;
+  shortName: string;
+  status: string;
+  startedAt: string;
+  analysisRun: AnalysisRunKind;
+};
+
+export type PodInfo = {
+  readyReplicas: number;
```

Comment:
In getReplicaSetInfo (down the line around lines 180-181), these fields are accessed with optional chaining
```suggestion
  readyReplicas?: number;
```